heuristic/alns.py

```diff
@@ -83,8 +85,63 @@ def __init__(self, state, data, criterion):
             self.shifts_per_week,
             self.t_covered_by_shift,
         )
+
+        remove_random_week = partial(
+            random_week_removal,
+            self.competencies,
+            self.employees,
+            self.weeks,
+            self.shifts_per_week,
+            self.t_covered_by_shift,
+            self.random_state,
+        )
+
+        remove_weighted_random_week = partial(
+            weighted_random_week_removal,
+            self.competencies,
+            self.time_periods_in_week,
+            self.combined_time_periods_in_week,
+            self.employees,
+            self.weeks,
+            self.L_C_D,
+            self.shifts_per_week,
+            self.t_covered_by_shift,
+            self.random_state,
+        )
+
+        remove_random_weekend = partial(
+            random_weekend_removal,
+            self.competencies,
+            self.employees,
+            self.weeks,
+            self.shifts_at_day,
+            self.t_covered_by_shift,
+            self.random_state,
+        )
+
         remove_worst_employee = partial(
-            worst_employee_removal, self.shifts, self.t_covered_by_shift, self.competencies
+            worst_employee_removal,
+            self.shifts,
+            self.t_covered_by_shift,
+            self.competencies,
+        )
+
+        remove_random_employee = partial(
+            random_employee_removal,
+            self.shifts,
+            self.t_covered_by_shift,
+            self.competencies,
+            self.employees,
+            self.random_state,
+        )
+
+        remove_weighted_random_employee = partial(
+            weighted_random_employee_removal,
+            self.shifts,
+            self.t_covered_by_shift,
+            self.competencies,
+            self.employees,
+            self.random_state,
         )
 
         repair_worst_week_regret = partial(
```

Comment:
Some sets are missing here that are included in daily_rest. Must be sure they are not overwritten with this merge.